frontend/openchat-client/src/openchat.ts

```diff
@@ -3492,9 +3592,18 @@ export class OpenChat extends OpenChatAgentWorker {
         if (newRole === oldRole) return Promise.resolve(true);
 
         // Update the local store
-        communityStateStore.updateProp(id, "members", (ps) =>
-            ps.map((p) => (p.userId === userId ? { ...p, role: newRole } : p)),
-        );
+        communityStateStore.updateProp(id, "members", (ms) => {
+            const userId = this._user?.userId;
```

Comment:
I don't get this. Why are we using _our own_ userId rather than the userId passed in?

Reply:
Oh dear! Copy paste error!